src/rtems/bench_porting_layer_rtems.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef PORTING_LAYER_RTEMS_H_
+#define PORTING_LAYER_RTEMS_H_
+
+/* includes */
+
+#include <stdint.h>
+#include <stddef.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include <stdbool.h>
+
+/* typedefs */
+
+typedef uint64_t bench_time_t;
+
+/* defines */
+
+#define PRINTF(FMT, ...)    printf(FMT, ##__VA_ARGS__)
+#define ARG_UNUSED(x)       (void)(x)
+#define BENCH_LAST_PRIORITY 10
+
+#define __weak __attribute__((__weak__))
+
+/*
+ * #define TICK_SYNC()    ....
+ * #define BENCH_IDLE_TIME
+ */
+
+/*
+ * Not all RTOSes support the same features. To help simplify the common code
+ * as much as possible, where there are known differences in support, various
+ * aspects of the benchmarking can be enabled/disabled as needed.
+ */
+
+#define RTOS_HAS_THREAD_SPAWN         1
+#define RTOS_HAS_THREAD_CREATE_START  0
+#define RTOS_HAS_SUSPEND_RESUME       0
+#define RTOS_HAS_MAIN_ENTRY_POINT     0
```

Comment:
Will we be adding kconfig at some point? =P

Reply:
I think that is a future work item. I've left it this way for the moment so that each supported RTOS configures these options in the same way.